pandas/core/groupby/groupby.py

```diff
@@ -2230,10 +2272,11 @@ def _get_cythonized_result(
         grouper = self.grouper
 
         labels, _, ngroups = grouper.group_info
-        output = collections.OrderedDict()  # type: dict
+        output = collections.OrderedDict()  # type: Dict[int, np.ndarray]
```

Comment:
Note that I typed this as Mapping originally but that yielded `pandas/core/groupby/groupby.py:2311: error: Unsupported target for indexed assignment`. I think not a big deal anyway since this is internal and literally is a Dict, but may be one difference between Mapping and Dict to consider going forward (cc @simonjayhawkins @TomAugspurger for related discussions)

Reply:
at least for the `attrs` case, we used Mapping for the *argument*, since any mapping is valid, and `Dict` for the actual value we store internally.

Reply:
yep. agree with Tom. Mapping for arguments, Dict for variables and return types.